### ISSUE TITLE
Use GitHub Personal Access token for hitting the API

### ISSUE THREAD

**Issue by raisedadead** [alexlafroscia/tldr-alfred#1]
I think we could use a personal access token to hit the API and overcome the rate limits, as per the guidelines? 

https://developer.github.com/v3/#rate-limiting

**Comment by alexlafroscia** [comment]
Good idea! It could be set as an environment variable in the Alfred workflow and used if available. I'm happy to take a PR, or will try this out myself if I get the change today.

**Comment by raisedadead** [comment]
Would love to contribute, here is an example https://github.com/stoe/alfred-github-tools, who seem to use token.

**Comment by alexlafroscia** [comment]
Now that I actually have reliable testing set up, I'm going to take a stab at this. It seems pretty easy to authenticate with the Github API using Basic Authentication with the username and a personal access token (as you suggested), and Node's HTTP request sending has an easy way for threading that through to the request.

Some related links (for my own sake, when I get back to this):

https://developer.github.com/v3/auth/#via-oauth-tokens
https://github.com/blog/1509-personal-api-tokens
https://nodejs.org/api/http.html#http_http_request_options_callback

**Comment by raisedadead** [comment]
@alexlafroscia awesome, thanks for fixing this so fast. 

I'll update the installation and let you know how this goes.
Cheers!

**Comment by alexlafroscia** [comment]
Awesome 👍 

Were you running into the rate limiting issue a lot?

**Comment by raisedadead** [comment]
> Were you running into the rate limiting issue a lot?

Nope, I did not. I just happened to checkout the Readme earlier, and thought that this could be an easy fix with the token.

This is method of auth is also used by some atom editor extensions (GFM markdown preview, or so I think).

So, far I don't think I have seen any errors (I have already added the secrets to my config).

Thanks once again for this awesome plugin.
